### ISSUE TITLE
Mozilla open sourced it's voice recognition model and data set.

### ISSUE THREAD

**Issue by LexiconCode** [dwks/silvius#6]
Mozilla released speech recognition model and voice data set that could be integrated. I'm curious to hear what people think.

https://blog.mozilla.org/blog/2017/11/29/announcing-the-initial-release-of-mozillas-open-source-speech-recognition-model-and-voice-dataset/

**Comment by dwks** [comment]
I fixed the link in your comment. There's also this more technical description which is very interesting:
https://hacks.mozilla.org/2017/11/a-journey-to-10-word-error-rate/

I've previously talked to some of the DeepSpeech devs and this seems like a good time to reach out again. It would be great to add DeepSpeech as a backend for Silvius. I suspect for good accuracy, we'd need to train it with custom command words as I did with Kaldi. If anyone has access to some GPU resources as part of a research group or something please let me know.

A lot of the DeepSpeech development is being driven by Mycroft (https://mycroft.ai/) as far as I know, so it will be interesting to see if they make a major announcement soon too.

**Comment by crypdick** [comment]
A few potentially helpful repos:

A GStreamer plug-in for DeepSpeech:
https://github.com/Elleo/gst-deepspeech

IBus plugin:
https://github.com/Elleo/ibus-deepspeech

**Comment by dwks** [comment]
Ah, thank you for the link to the GStreamer plug-in. That will definitely help Silvius use DeepSpeech. However, I'm not convinced that DeepSpeech is a good fit for voice coding; its word error rate is higher than the existing model, and it's unlikely to be as good at recognizing individual command words. Will probably need more training data in the form of commands for that. It's still worth a try though; I should have time in a few weeks to look at this.

**Comment by LexiconCode** [comment]
https://github.com/pykaldi/pykaldi

**Comment by crypdick** [comment]
@dwks when you say DeepSpeech's word error rate is higher, are you referring to Kaldi's Gaussian Mixture Model?

**Comment by dwks** [comment]
I believe I was referring to the published word error rate for librispeech.
The model silvius is currently using is tedlium, but also a modified
tedlium boosted for command words (which probably decreases the overall
accuracy). We can certainly do better with librispeech or the newest aspire
models (which are nnet3). Unfortunately I think some code needs to be
written to integrate the current streaming system with neural nets (nnet2,
nnet3).
